WordPress/gutenberg
# Allow explicitly specifying additional block navigation menu options in block edit function

## Description

This PR is a Proof of Concept of explicitly specifying per-block ellipsis menu items in the edit.js file.

This is a part of a sequence of few PRs.  I propose merging/reviewing these PRs in the following order:

1. ~~#22427 - introduce very basic ellipsis menu to the block navigation~~
1. #22463 - allow the block to define additional ellipsis menu items
1. #22462 - introduce an abstraction that takes care of the toolbar items and ellipsis menu items at the same time

## How has this been tested?
1. Go to the experimental navigation screen
1. Open the ellipsis menu
1. Confirm the Add Submenu is there

Note that adding a submenu transfer focus to the Navigation menu page - this is something I am looking forward to addressing in a separate PR once the slot mechanics from this PR are in place.

## Screenshots

<img width="930" alt="Zrzut ekranu 2020-05-21 o 11 34 14" src="https://user-images.githubusercontent.com/205419/82545471-05346000-9b57-11ea-8feb-470925583a2e.png">

## Types of changes
New feature (non-breaking change which adds functionality)
